src/Util.h

```diff
@@ -348,6 +348,15 @@ bool sub_would_overflow(int bits, int64_t a, int64_t b);
 bool mul_would_overflow(int bits, int64_t a, int64_t b);
 // @}
 
+/** Routines to perform arithmetic on signed types without triggering signed
+ * overflow. If overflow would occur, sets result to zero, and returns
+ * false. Otherwise set result to the correct value, and returns true. */
+// @{
+bool add_with_overflow(int bits, int64_t a, int64_t b, int64_t *result);
```

Comment:
Since it looks like almost every call to these uses the result, probably worthwhile to mark with with MUST_USE_RESULT so that the exceptions are more obvious and explicit